From d47d780c837a422fc4629be91b01a72e0d19960c Mon Sep 17 00:00:00 2001
From: Martin Grund <mgrund@cloudera.com>
Date: Fri, 28 Aug 2015 16:25:59 -0700
Subject: [PATCH] IMPALA-1136: Skip \u0000 characters when dealing Avro schemas

The limitation of the Avro JSON library not to handle \u0000 characters
is to avoid problems with builtin functions like strlen() that would
report wrong length when encountering such a character. Now, in the case
of Impala, for now, we don't support any Unicode characters. This allows
us to actually skip the \u0000 character instead of interpreting it.

It is important to say that even the most recent versions of Avro do not
support parsing \u0000 characters.

Change-Id: I56dfa7f0f12979fe9705c51c751513aebce4beca

diff --git a/thirdparty/avro-c-1.7.4/jansson/src/load.c b/thirdparty/avro-c-1.7.4/jansson/src/load.c
index 338f3c1..67092cf 100644
--- a/thirdparty/avro-c-1.7.4/jansson/src/load.c
+++ b/thirdparty/avro-c-1.7.4/jansson/src/load.c
@@ -400,8 +400,7 @@ static void lex_scan_string(lex_t *lex, json_error_t *error)
                 }
                 else if(value == 0)
                 {
-                    error_set(error, lex, "\\u0000 is not allowed");
-                    goto out;
+                    continue;
                 }

                 if(utf8_encode(value, buffer, &length))
--
2.3.5
